backend/src/types.ts

```diff
@@ -219,3 +219,25 @@ export type BuildStatus = {
   status: BUILD_PHASE;
   timestamp?: string;
 };
+
+export enum NotebookStatus {
+  VALIDITING,
+  SUCCESS,
+  FAILED
+}
+
+export type Notebook = {
+  name: string;
+  repo: string;
```

Comment:
@dlabaj can we revisit this? Is this schema based on https://studio.apicur.io/preview?aid=66775#operation/getNotebooks ?

![image](https://user-images.githubusercontent.com/7453394/159479879-3e1a7bd0-18d1-468f-a368-06f70a2f80cb.png)

I think this doesn't match, because:

1. We have no idea about a `repo`. We import from container image repository not github repository. Can we call this `url`?
2. Can we change `status` to `phase` so it matches the backend names better?

Reply:
For item 1 we can definitely change it to "URL" Should we update mocks ups so they say URL as well?   https://marvelapp.com/prototype/6dcfc3h/screen/85139860

For item 2 it should have been phase.   Status from this mockup I am assuming would show the phase while it's being imported and then change to the visible setting once it reaches success?  https://marvelapp.com/prototype/6dcfc3h/screen/85139859
@tumido Let me know if that's your understanding as well.

Reply:
> For item 1

I think this is a tricky part on the UI. Imagine this example. You go to URL `https://quay.io/repository/tcoufal/jh-minimal-test`. This image can be pulled by podman/docker as `quay.io/tcoufal/jh-minimal-test` URI which in container terminology (see https://developers.redhat.com/blog/2018/02/22/container-terminology-practical-introduction#repository) is referred to as a "repository"... 

![image](https://user-images.githubusercontent.com/7453394/159531197-50f98aa3-662c-4e25-b462-0ee7cb21c039.png)

If we name it URL in the UI, we need to clarify to the user we want the  `quay.io/tcoufal/jh-minimal-test` value, not `https://quay.io/repository/tcoufal/jh-minimal-test`. On the other hand, naming it "Repository" carries the "code repository" connotation (as seen in the API example)...

In addition to that, we have an annotation in JupyterHub ImageStreams called `opendatahub.io/notebook-image-url`. This annotation is used to point to the git repo where the source code for the image lives. Again, using the `url` and `repository` as synonyms...

In the BYON pipelines, we call the parameter URL. 

Honestly the more I think about it, the more confused I am... I think I'm overcomplicating things but I would really like to have things named consistently. No matter which term we end up using.

> For item 2

Honestly, I think this is a common confusion and inconsistency in OpenShift. Take Pods as an example:

![image](https://user-images.githubusercontent.com/7453394/159529791-21c6dd8e-0374-4931-b047-bfde5429a74b.png)

The UI shows column named `Status` but API serves the value in `.status.phase` field...

In our case we're gonna source this value from `opendatahub.io/notebook-image-phase` annotation. So... I think for the sake of UI consistency the column should be definitely called "Status" (user expect that based on the rest of OpenShift UI). When it comes to the API side, the custom is to call this value a `phase`, since `status` usually means a complex field containing many different status-related information.